services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -468,6 +414,77 @@ async def upload_file_to_datcore(
 
         # actually we have to query the master db
 
+    async def metadata_file_updater(
+        self,
+        file_uuid: str,
+        bucket_name: str,
+        object_name: str,
+        file_size: int,
+        last_modified: str,
+        max_update_retries: int = 50,
+    ):
+        """
+        Will retry max_update_retries to update the metadata on the file after an upload.
+        If it is not successfull it will exit and log an error.
```

Comment:
why not use tenacity here? I think most of these features are available in the @retry decorator.

Reply:
Because it was too generic for my use case. I needed to await only for a subsection of a function.